pallets/settlement/src/benchmarking.rs

```diff
@@ -473,94 +786,139 @@ benchmarks! {
+        // Emulate the add instruction and get all the necessary arguments.
+        let (legs, venue_id, origin, did , s_portfolios, r_portfolios, account_id) = emulate_add_instruction::<T>(1, true)?;
+        // Add instruction
+        Module::<T>::base_add_instruction(did, venue_id, SettlementType::SettleOnAffirmation, None, legs.clone())?;
+        let instruction_id = 1;
+        let ticker = Ticker::try_from(vec![b'A'; 1 as usize].as_slice()).unwrap();
+        let receipt = create_receipt_details::<T>(0, legs.first().unwrap().clone());
+        let leg_id = 0;
+        let amount = 100;
+        // Some manual setup to support the extrinsic.
+        set_instruction_leg_status_to_pending::<T>(instruction_id, leg_id);
+        T::Portfolio::lock_tokens(s_portfolios.first().unwrap(), &ticker, &amount.into())?;
+    }: _(origin, instruction_id, receipt.clone())
+    verify {
+        ensure!(Module::<T>::instruction_leg_status(instruction_id, leg_id) ==  LegStatus::ExecutionToBeSkipped(
+            receipt.signer,
+            receipt.receipt_uid,
+        ), "Settlement: Fail to unclaim the receipt");
+    }
+
+
+    affirm_with_receipts {
+        // Catalyst here is the length of receipts vector.
+        let r in 1 .. T::MaxLegsInInstruction::get() as u32;
+        // Emulate the add instruction and get all the necessary arguments.
+        let (legs, venue_id, origin, did , s_portfolios, r_portfolios, account_id) = emulate_add_instruction::<T>(r, true)?;
+        // Add instruction
+        Module::<T>::base_add_instruction(did, venue_id, SettlementType::SettleOnAffirmation, None, legs.clone())?;
+        let instruction_id = 1;
+        let mut receipt_details = Vec::with_capacity(r as usize);
+        legs.clone().into_iter().enumerate().for_each(|(idx, l)| {
+            receipt_details.push(create_receipt_details::<T>(idx as u32, l));
+        });
+    }: _(origin, instruction_id, receipt_details.clone(), s_portfolios)
+    verify {
+        for (i, receipt) in receipt_details.iter().enumerate() {
+            ensure!(Module::<T>::instruction_leg_status(instruction_id, i as u64) ==  LegStatus::ExecutionToBeSkipped(
+                receipt.signer.clone(),
+                receipt.receipt_uid,
+            ), "Settlement: Fail to affirm with receipts");
+        }
+    }
+
+    execute_scheduled_instruction {
+        // This dispatch execute an instruction.
+        //
+        // Worst case scenarios.
+        // 1. Create maximum legs and both traded assets are different assets/securities.
+        // 2. Assets should have worst compliance restrictions ?
+        // 3. Assets have maximum no. of TMs.
+
+        let l in 2 .. T::MaxLegsInInstruction::get() as u32; // At least 2 legs needed to achieve worst case.
+        let s in 0 .. T::MaxNumberOfTMExtensionForAsset::get() as u32;
+        let c in 1 .. MAX_COMPLIANCE_RESTRICTION; // At least 1 compliance restriction needed.
```

Comment:
The same point as above, I don't see why 1 compliance is required?

Reply:
1 compliance is required to pass the transfer otherwise it will fail so I don't see a point to start it with 0.